### ISSUE TITLE
User notifications (email)

### ISSUE THREAD

**Issue by sballesteros** [PREreview/rapid-prereview#2]
**Updated per discussion with @dasaderi and @SamanthaHindle   on slack**

## User notifications (via email)

- [ ] when a user review has been moderated (note that this is triggered only _after_ moderators validated user reports and not each time a user reports a potential issue).
- [ ] when a user has been blocked (lose write access to the platform) by a moderator.
- [ ] when a review is added to a preprint for which the user had requested feedback (via a request for reviews).

### Backend

- [x] Get a default email address from ORCID (see https://members.orcid.org/api/tutorial/read-orcid-records )
- [ ] Allow user to update that email (or set it in case where no emails could be found in the public ORCID record)
- [ ] Add logic to verify the email address (only for emails not imported from ORCID). Notifications will only be sent to emails that are verified. Verification will be done by sending an email with a "magic" link to the user. If the user receives the email and clicks on the link the email address will be considered verified.
- [ ] Add ability to set notification preferences (see frontend section below)
- [ ] Send the emails with sendgrid on Azure. **Emails will be sent as plain text only (so unstyled).**

### Frontend
- [ ] UI to display and update the email address (part of the settings). Note that the email address is _never_ displayed publicly and is only used for notification purposes.
- [ ] UI to keep track of the verification status of an email
- [ ] UI for the notification preference settings (YES / NO) to assess whether or not notifications should be sent when new reviews are added to a preprint for which the user requested feedback

**Comment by sballesteros** [comment]
I really like the distinction made by @dasaderi here:

> If I am a user I want to know if the authors have explicitly asked for feedback and I'm going to value that request more than if another community member has asked for feedback.

However this requires significantly more work data that we currently have (knowing when a user is actually author of a preprint). @dasaderi maybe you can create another issue for that that can be tackled later once more work is done to have a good understanding of who are the preprint authors?
Otherwise we probably need to prioritize that work before doing the notification part.

**Comment by sballesteros** [comment]
```js
const updateEmailAction = {
  '@type': 'UpdateEmailAction',
  agent: 'user:userId',
  object: 'user:userId',
  payload: {
    email: 'mailto:me@example.com'
  },
  result: {
    '@id': 'user:userId',
    contactPoint: {
      '@type': 'ContactPoint',
      contactType: 'notifications',
      email: 'mailto:me@example.com',
      token: {
        '@type': 'EmailVerificationToken',
        value: 'XXXX'
      }
    }
  }
};

const verifyEmailAction = {
  '@type': 'VerifyEmailAction',
  agent: 'user:userId',
  object: 'mailto:me@example.com',
  token: {
    '@type': 'EmailVerificationToken',
    value: 'XXXX'
  },
  result: {
    '@id': 'user:userId',
    contactPoint: {
      '@type': 'ContactPoint',
      contactType: 'notifications',
      email: 'mailto:me@example.com',
      dateEmailVerified: new Date().toISOString()
    }
  }
};
```
